salt/haproxy/haproxy.cfg.jinja

```diff
@@ -0,0 +1,28 @@
+global
+        log /dev/log    local0
+        log /dev/log    local1 notice
+        daemon
+
+defaults
+        log     global
+        mode    tcp
+        option  tcplog
+        option  dontlognull
+        timeout connect 5000
+        timeout client 50000
+        timeout server 50000
+
+listen kubernetes-master
+        bind 127.0.0.1:6444
+        mode tcp
+        default-server inter 10s fall 3
+        balance roundrobin
+
+{%- for minion_id, addrlist in salt['mine.get']('roles:kube-master', 'network.ip_addrs', 'grain').items() %}
+{% if addrlist is string -%}
+{% set addr = addrlist -%}
+{% else -%}
+{% set addr = addrlist|first -%}
+{% endif -%}
+        server master-{{ minion_id }} {{ addr }}:6443 check
+{% endfor -%}
```

Comment:
Something to think in the future: what happens if I add a new node and I want it to become a master, will this work ? In the past we've been using [confd](https://github.com/kelseyhightower/confd) to handle this, but maybe this is something that salt can handle as well ?